test/test_nn.py

```diff
@@ -4238,6 +4238,35 @@ def test_CTCLoss_cudnn(self):
         self.assertEqual(res, expected)
         self.assertEqual(res2, res)
 
+    def test_CTCLoss_typechecks(self):
+        target_lengths = torch.tensor([30, 25, 20])
+        input_lengths = torch.tensor([50, 50, 50])
+        targets = torch.randint(1, 15, (sum(target_lengths),), dtype=torch.int)
+        log_probs = torch.randn(50, 3, 15, dtype=torch.float).log_softmax(2)
+        with self.assertRaises(RuntimeError):
+            _input_lengths = input_lengths.to(dtype=torch.float)
+            torch.nn.functional.ctc_loss(log_probs, targets, _input_lengths, target_lengths)
+        with self.assertRaises(RuntimeError):
+            target_lengths = target_lengths.to(dtype=torch.float)
+            torch.nn.functional.ctc_loss(log_probs, targets, input_lengths, target_lengths)
+
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    def test_CTCLoss_lengthchecks_cuda(self):
+        target_lengths = [30, 25, 20]
+        input_lengths = [50, 50, 50]
+        targets = torch.randint(1, 15, (3, 29), dtype=torch.int).to(dtype=torch.cuda.FloatTensor)
+        log_probs = torch.randn(50, 3, 15, dtype=torch.float).log_softmax(2).to(dtype=torch.cuda.FloatTensor)
```

Comment:
You'd want `.to(dtype=torch.float, device='cuda')`, or really
`targets = torch.randint(1, 15, (3, 29), dtype=torch.long, device='cuda')` (targets should be on cuda but long to hit the native version).
Similarly, you'd just add `device=cuda` to the randn and drop the `.to(...)`.